app/views/play/level/PlayLevelView.coffee

```diff
@@ -196,18 +205,53 @@ module.exports = class PlayLevelView extends RootView
         indefiniteLength: e.level.isType('game-dev')
       })
     @setupGod() if @waitingToSetUpGod
+    @levelSlug = e.level.get('slug')
+
+  checkTournamentEnd: =>
+    return unless @timeOff
+    return unless @mandate.loaded
+    return unless @levelSlug
+    courseInstanceID = @courseInstanceID or utils.getQueryVariable 'league'
+    mandate = @mandate.get('0')
+
+    tournamentState = 1
+    #        tournamentState table
+    #  ladder\checkTournamentEnd   keep    stop
+    #  open                         3       2
+    #  close                        1       0
+    if mandate
+      tournamentState = @getTournamentState mandate, courseInstanceID, @levelSlug, @timeOff
+      unless @tournamentEnd or me.isAdmin() or tournamentState > 1
+        window.location.href = '/play/ladder/'+@levelSlug+(if courseInstanceID then '/course/'+courseInstanceID else "")
+    if tournamentState % 2 == 0
+      clearInterval @checkTournamentEndInterval
+    return @checkTournamentEnd
+
+  getTournamentState: (mandate, courseInstanceID, levelSlug, timeOff) ->
+    tournament = _.find mandate.currentTournament or [], (t) =>
+      t.courseInstanceID is courseInstanceID and t.level is levelSlug
+    if tournament
+      currentTime = (Date.now() + timeOff) / 1000
+      console.log "Current time:", new Date(currentTime * 1000)
+      if currentTime < tournament.startAt
+        delta = tournament.startAt - currentTime
+        console.log "Tournament will start at: #{new Date(tournament.startAt * 1000)}, Time left: #{parseInt(delta / 60 / 60) }:#{parseInt(delta / 60) % 60}:#{parseInt(delta) % 60}"
+        return 1
```

Comment:
Also same comment as above. Better if this returns a named constant instead of a literal.